cmd/parca-agent/main.go

```diff
@@ -103,6 +105,16 @@ func main() {
 
 	logger := logger.NewLogger(flags.LogLevel, logger.LogFormatLogfmt, "")
 
+	// Fetch build info such as the git revision we are based off
+	buildInfo, err := buildinfo.FetchBuildInfo()
```

Comment:
Overall, this PR looks good. It would help us to discover  build information during runtime.

However, those variables are set by our build tool from outside while cross-building the binaries. https://github.com/parca-dev/parca-agent/blob/ff872f449ea39ae9d37988bbc4162e905f955db0/.goreleaser.yml#L26

So we should check if those variables have already values before we set them.